tstest/integration: flake in race builder, logtail uploaded zero bytes?

See at https://github.com/tailscale/tailscale/runs/2917864453, with git parent 194d5b841266f9b7b8e61da758582925d289ec7e

```
--- FAIL: TestOneNodeUp_NoAuth (25.90s)
Error:     integration.go:66: built [tailscale.com/cmd/tailscaled tailscale.com/cmd/tailscale] in 1.75s
Error:     integration_test.go:75: Status: NeedsLogin
Error:     integration_test.go:84: log catcher didn't see `Program starting: `; got logtail started
        error from POST of "{\"text\": \"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00{\\\"logtail\\\": {\\\"client_time\\\": \\\"2021-06-25T21:19:28.877304907Z\\\"}, \\\"text\\\": \\\"logtail: dialed \\\\\\\"127.0.0.1:46775\\\\\\\" in 16ms\\\\n\\\"}\\n\"}\n": invalid character '\x00' in string literal
        netmap packet filter: (not ready yet)
        Start
        using backend prefs
        created empty state for "_daemon": Prefs{ra=true dns=true want=false routes=[] nf=on Persist=nil}
        got initial portlist info in 0s
        magicsock: disco key = d:80830d8daeeef41e
        control: HostInfo: {"IPNVersion":"date.20210603","BackendLogID":"d0f3dad19f9b51abef23db79dc66fc2d48b7f08563c7946c79c2cc887e4c76ed","OS":"linux","OSVersion":"Ubuntu 20.04.2 LTS (Focal Fossa); kernel=5.8.0-1033-azure","Hostname":"fv-az210-874","GoArch":"amd64","Services":[{"Proto":"tcp","Port":22},{"Proto":"tcp","Port":53},{"Proto":"tcp","Port":8084},{"Proto":"tcp","Port":58436,"Description":"tailscaled"}]}
        control: mapRoutine: state:new
        control: authRoutine: state:new; goal=nil paused=false
        Backend: logs: be:d0f3dad19f9b51abef23db79dc66fc2d48b7f08563c7946c79c2cc887e4c76ed fe:
        Switching ipn state NoState -> NeedsLogin (WantRunning=false, nm=false)
        blockEngineUpdates(true)
        health("overall"): error: state=NeedsLogin, wantRunning=false
        wgengine: Reconfig: configuring userspace wireguard config (with 0/0 peers)
        wgengine: Reconfig: configuring router
        warning: fakeRouter.Set: not implemented.
        wgengine: Reconfig: configuring DNS
        dns: Set: {DefaultResolvers:[] Routes:map[] SearchDomains:[] Hosts:map[]}
        dns: Resolvercfg: {Routes:map[] Hosts:map[] LocalDomains:[]}
        dns: OScfg: {Nameservers:[] SearchDomains:[] MatchDomains:[]}
        wgengine: Reconfig done
        ipnserver: conn1: incoming control connection
        ipnserver: conn1: connection from userid 1001; connection from non-root user matching daemon has access
        ipnserver: conn1: ReadMsg: EOF
        control: HostInfo: {"IPNVersion":"date.20210603","BackendLogID":"d0f3dad19f9b51abef23db79dc66fc2d48b7f08563c7946c79c2cc887e4c76ed","OS":"linux","OSVersion":"Ubuntu 20.04.2 LTS (Focal Fossa); kernel=5.8.0-1033-azure","Hostname":"fv-az210-874","GoArch":"amd64"}
        control: cancelMapSafely: synced=false
        control: client.Shutdown()
        control: client.Shutdown: inSendStatus=0
        control: authRoutine: context done.
        control: authRoutine: state:synchronized; goal=nil paused=false
        [RATELIMIT] format("control: authRoutine: %s; goal=nil paused=%v")
        control: authRoutine: quit
        control: mapRoutine: quit
        control: Client.Shutdown done.
        magicsock: closing connection to derp-1 (conn-close), age 0s
        magicsock: 0 active derp conns
        Routine: receive incoming v6 - stopped
        warning: fakeRouter.Close: not implemented.
        Device closing
        Routine: receive incoming v4 - stopped
        Routine: receive incoming receiveDERP - stopped
        Device closed
        flushing log.
        logger closing down
        control: cancelMapSafely: wrote to channel
        control: mapRoutine: new map needed while idle.
        control: mapRoutine: state:new
        magicsock: ignoring pre-DERP map, STUN-less endpoint update: [{10.1.0.168:38937 local} {172.17.0.1:38937 local}]
        
Error:     integration_test.go:421: Running up --login-server=http://127.0.0.1:37375 ...
Error:     integration_test.go:94: Got IP: 100.64.0.1
Error:     integration_test.go:99: number of HTTP logcatcher requests: 52
Error:     stuntest.go:64: STUN server shutdown
FAIL
exit status 1
FAIL	tailscale.com/tstest/integration	39.124s
```

Again:

```
--- FAIL: TestNoControlConnectionWhenDown (29.03s)
Error:     integration.go:74: built [tailscale.com/cmd/tailscaled tailscale.com/cmd/tailscale] in 2.172s
Error:     integration_test.go:328: Status: NeedsLogin
Error:     integration_test.go:659: Running [up --login-server=http://127.0.0.1:37367] ...
Error:     integration_test.go:667: Running down ...
Error:     integration_test.go:342: Status: Stopped
Error:     integration_test.go:342: log catcher didn't see `Program starting: `; got netmap packet filter: (not ready yet)
        Start
        using backend prefs
        created empty state for "_daemon": Prefs{ra=true dns=true want=false routes=[] nf=on Persist=nil}
        Creating wireguard device...
        Bringing wireguard device up...
        UDP bind has been updated
        Interface state was Down, requested Up, now Up
        Bringing router up...
        warning: fakeRouter.Up: not implemented.
        Clearing router settings...
        warning: fakeRouter.Set: not implemented.
        Starting link monitor...
        Starting magicsock...
        Routine: receive incoming v4 - started
        Routine: receive incoming v6 - started
        Routine: receive incoming receiveDERP - started
        Engine created.
        magicsock: starting endpoint update (initial)
        magicsock: ignoring pre-DERP map, STUN-less endpoint update: [{10.1.1.67:45613 local} {172.17.0.1:45613 local}]
        Listening on /tmp/TestNoControlConnectionWhenDown3089177389/002/tailscale.sock
        ipnserver: conn3: incoming control connection
        ipnserver: conn3: connection from userid 1001; connection from non-root user matching daemon has access
        netmap packet filter: (not ready yet)
        Start
        using backend prefs
        backend prefs for "_daemon": Prefs{ra=false dns=true want=false routes=[] nf=on url="http://127.0.0.1:37367" Persist{lm=, o=, n=[rEQRO] u="user-1@fake-control.example.net"}}
        got initial portlist info in 0s
        magicsock: disco key = d:99dcff515ac32176
        control: HostInfo: {"IPNVersion":"date.20210603","BackendLogID":"379b3ab19ca12a9e9919a1f23d05646f0dbc9bac76ea7482c5f2d8abc4012964","OS":"linux","OSVersion":"Ubuntu 20.04.2 LTS (Focal Fossa); kernel=5.8.0-1036-azure","Hostname":"fv-az210-50","GoArch":"amd64","Services":[{"Proto":"tcp","Port":22},{"Proto":"tcp","Port":53},{"Proto":"tcp","Port":8084}]}
        control: authRoutine: state:new; goal=nil paused=false
        control: authRoutine: state:new; goal=nil paused=false
        control: authRoutine: quit
        control: mapRoutine: context done.
        control: mapRoutine: state:new
        control: mapRoutine: quit
        control: Client.Shutdown done.
        using backend prefs
        created empty state for "_daemon": Prefs{ra=true dns=true want=false routes=[] nf=on Persist=nil}
        error from POST of "[{\"text\": \"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00{\\\"logtail\\\": {\\\"client_time\\\": \\\"2021-07-20T20:35:15.571694481Z\\\"}, \\\"text\\\": \\\"generating new machine key\\\\n\\\"}\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.5731493Z\"}, \"text\": \"machine key written to store\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.573882209Z\"}, \"text\": \"control: HostInfo: {\\\"IPNVersion\\\":\\\"date.20210603\\\",\\\"BackendLogID\\\":\\\"379b3ab19ca12a9e9919a1f23d05646f0dbc9bac76ea7482c5f2d8abc4012964\\\",\\\"OS\\\":\\\"linux\\\",\\\"OSVersion\\\":\\\"Ubuntu 20.04.2 LTS (Focal Fossa); kernel=5.8.0-1036-azure\\\",\\\"Hostname\\\":\\\"fv-az210-50\\\",\\\"GoArch\\\":\\\"amd64\\\",\\\"Services\\\":[{\\\"Proto\\\":\\\"tcp\\\",\\\"Port\\\":22},{\\\"Proto\\\":\\\"tcp\\\",\\\"Port\\\":53},{\\\"Proto\\\":\\\"tcp\\\",\\\"Port\\\":8084}]}\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.574251114Z\"}, \"text\": \"control: mapRoutine: state:new\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.574399316Z\"}, \"text\": \"control: authRoutine: state:new; goal=nil paused=false\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.574483717Z\"}, \"text\": \"Backend: logs: be:379b3ab19ca12a9e9919a1f23d05646f0dbc9bac76ea7482c5f2d8abc4012964 fe:\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.57470622Z\"}, \"text\": \"Switching ipn state NoState -> NeedsLogin (WantRunning=true, nm=false)\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.574816121Z\"}, \"text\": \"blockEngineUpdates(true)\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.575136825Z\"}, \"text\": \"wgengine: Reconfig done\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.57550303Z\"}, \"text\": \"StartLoginInteractive: url=false\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.575584131Z\"}, \"text\": \"control: client.Login(false, 2)\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.575654132Z\"}, \"text\": \"control: authRoutine: context done.\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.575747233Z\"}, \"text\": \"control: authRoutine: state:new; wantLoggedIn=true\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.575831634Z\"}, \"text\": \"control: direct.TryLogin(token=false, flags=2)\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.575925435Z\"}, \"text\": \"control: LoginInteractive -> regen=true\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.576096538Z\"}, \"text\": \"control: doLogin(regen=true, hasUrl=false)\\n\"}\n]": invalid character '\x00' in string literal
        control: control server key cb80a72eeb02acdc71059dcae6cf412640d46f8a4cb64ed20c7a6761c614bb28 from http://127.0.0.1:38777
        control: Generating a new nodekey.
        control: RegisterReq: onode=[AAAAA] node=[J9/bz] fup=false
        "2021-07-20T20:35:15.583370631Z"}, "text": "control: authRoutine: context done.\n"}
        control: authRoutine: state:new; wantLoggedIn=true
        control: direct.TryLogin(token=false, flags=0)
        control: doLogin(regen=false, hasUrl=false)
        health("overall"): error: state=Stopped, wantRunning=false
        wgengine: Reconfig: configuring userspace wireguard config (with 0/0 peers)
        wgengine: Reconfig: configuring router
        warning: fakeRouter.Set: not implemented.
        wgengine: Reconfig: configuring DNS
        dns: Set: {DefaultResolvers:[] Routes:{} SearchDomains:[] Hosts:0}
        dns: Resolvercfg: {Routes:{} Hosts:0 LocalDomains:[]}
        dns: OScfg: {Nameservers:[] SearchDomains:[] MatchDomains:[]}
        wgengine: Reconfig done
        control: control server key 5ddb0f9dac923b2731d127f5947a8328a89582fc32271725750e5f61c1ece715 from http://127.0.0.1:37367
        control: RegisterReq: onode=[AAAAA] node=[rEQRO] fup=false
        control: RegisterReq: got response; nodeKeyExpired=false, machineAuthorized=true; authURL=false
        control: No AuthURL
        control: sendStatus: authRoutine-success: state:authenticated
        blockEngineUpdates(false)
        authReconfig: netmap not yet valid. Skipping.
        authReconfig: netmap not yet valid. Skipping.
        control: cancelMapSafely: synced=false
        control: cancelMapSafely: wrote to channel
        control: authRoutine: state:authenticated; goal=nil paused=false
        control: mapRoutine: new map needed while idle.
        control: mapRoutine: state:authenticated
        control: PollNetMap: stream=true :0 ep=[]
        control: RegisterReq: got response; nodeKeyExpired=false, machineAuthorized=true; authURL=false
        control: No AuthURL
        control: sendStatus: authRoutine-success: state:authenticated
        authReconfig: netmap not yet valid. Skipping.
        control: cancelMapSafely: synced=false
        control: mapRoutine: new map needed while idle.
        control: cancelMapSafely: wrote to channel
        control: mapRoutine: state:authenticated
        control: PollNetMap: stream=true :0 ep=[]
        control: authRoutine: state:authenticated; goal=nil paused=false
        control: new network map[0]:
        netmap: self: [J9/bz] auth=machine-authorized u=userid:1 debug={"DisableUPnP":true} [100.64.0.1/32 fd7a:115c:a1e0:ab12:4843:cd96:6240:1/128]
        control: mapRoutine: netmap received: state:synchronized
        control: sendStatus: mapRoutine-got-netmap: state:synchronized
        active login: <missing-profile>
        netmap packet filter: [[TCP UDP ICMPv4 ICMPv6][0.0.0.0/0,::/0]=>[0.0.0.0/0:*,::/0:*]]
        magicsock: got updated network map; 0 peers (0 with discokey)
        netstack: deregistered IP 255.255.255.255/32
        netstack: registered IP 100.64.0.1/32
        netstack: registered IP fd7a:115c:a1e0:ab12:4843:cd96:6240:1/128
        magicsock: starting endpoint update (derp-map-update)
        error from POST of "[{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.636293511Z\"}, \"text\": \"control: new network map[0]:\\nnetmap: self: [rEQRO] auth=machine-authorized u=userid:1 debug={\\\"DisableUPnP\\\":true} [100.64.0.1/32 fd7a:115c:a1e0:ab12:4843:cd96:6240:1/128]\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.636457713Z\"}, \"text\": \"control: mapRoutine: netmap received: state:synchronized\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.636641816Z\"}, \"text\": \"control: sendStatus: mapRoutine-got-netmap: state:synchronized\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.636817118Z\"}, \"text\": \"active login: <missing-profile>\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.63696932Z\"}, \"text\": \"control: setPaused(true)\\n\"}\n,{\"text\": \"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00{\\\"logtail\\\": {\\\"client_time\\\": \\\"2021-07-20T20:35:15.632961168Z\\\"}, \\\"text\\\": \\\"Switching ipn state NeedsLogin -> Starting (WantRunning=true, nm=true)\\\\n\\\"}\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.63462699Z\"}, \"text\": \"magicsock: SetPrivateKey called (init)\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.634883093Z\"}, \"text\": \"wgengine: Reconfig: configuring userspace wireguard config (with 0/0 peers)\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.635088896Z\"}, \"text\": \"magicsock: ReSTUN: endpoint update active, need another later (\\\"set-private-key\\\")\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.635589902Z\"}, \"text\": \"UAPI: Updating private key\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.63697232Z\"}, \"text\": \"wgengine: Reconfig: configuring router\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.63701632Z\"}, \"text\": \"warning: fakeRouter.Set: not implemented.\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.637058121Z\"}, \"text\": \"wgengine: Reconfig: configuring DNS\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.637133622Z\"}, \"text\": \"dns: Set: {DefaultResolvers:[] Routes:{} SearchDomains:[] Hosts:0}\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.637209823Z\"}, \"text\": \"dns: Resolvercfg: {Routes:{} Hosts:0 LocalDomains:[]}\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.637266324Z\"}, \"text\": \"dns: OScfg: {Nameservers:[] SearchDomains:[] MatchDomains:[]}\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.637303824Z\"}, \"text\": \"wgengine: Reconfig done\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.637369225Z\"}, \"text\": \"authReconfig: ra=false dns=true 0x01: <nil>\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.637542527Z\"}, \"text\": \"peerapi: serving on http://100.64.0.1:58436\\n\"}\n,{\"logtail\": {\"client_time\": \"2021-07-20T20:35:15.637664429Z\"}, \"text\": \"peerapi: serving on http://[fd7a:115c:a1e0:ab12:4843:cd96:6240:1]:58436\\n\"}\n]": invalid character '\x00' in string literal
        control: mapRoutine: state:authenticated
        [RATELIMIT] format("control: mapRoutine: %s")
        control: PollNetMap: stream=true :0 ep=[]
        control: successful lite map update in 12ms
        ipnserver: conn1: incoming control connection
        ipnserver: conn1: connection from userid 1001; connection from non-root user matching daemon has access
        ipnserver: conn1: ReadMsg: EOF
        control: HostInfo: {"IPNVersion":"date.20210603","BackendLogID":"379b3ab19ca12a9e9919a1f23d05646f0dbc9bac76ea7482c5f2d8abc4012964","OS":"linux","OSVersion":"Ubuntu 20.04.2 LTS (Focal Fossa); kernel=5.8.0-1036-azure","Hostname":"fv-az210-50","GoArch":"amd64","Services":[{"Proto":"tcp","Port":22},{"Proto":"tcp","Port":53},{"Proto":"tcp","Port":8084},{"Proto":"tcp","Port":46590,"Description":"tailscaled"},{"Proto":"tcp","Port":58436,"Description":"tailscaled"}]}
        control: cancelMapSafely: synced=false
        control: cancelMapSafely: wrote to channel
        netcheck: report: udp=false v4=false v6=false mapvarydest= hair= portmap= derp=0
        magicsock: netInfo update: NetInfo{varies= hairpin= ipv6=false udp=false derp=#1 portmap= link=""}
        magicsock: endpoints changed: 10.1.1.67:45613 (local), 172.17.0.1:45613 (local)
        control: NetInfo: NetInfo{varies= hairpin= ipv6=false udp=false derp=#1 portmap= link=""}
        control: cancelMapSafely: synced=false
        control: cancelMapSafely: wrote to channel
        control: client.newEndpoints(0, [10.1.1.67:45678 172.17.0.1:45678])
        control: cancelMapSafely: synced=false
        control: cancelMapSafely: channel was full
        magicsock: starting endpoint update (non-zero-peers)
        derphttp.Client.Connect: connecting to derp-1 (test)
        Switching ipn state Starting -> Running (WantRunning=true, nm=true)
        control: client.newEndpoints(0, [10.1.1.67:45613 172.17.0.1:45613])
        control: cancelMapSafely: synced=false
        control: cancelMapSafely: channel was full
```

cc @josharian

--- a/logpolicy/logpolicy.go
+++ b/logpolicy/logpolicy.go
@@ -128,6 +128,13 @@ func (l logWriter) Write(buf []byte) (int, error) {
 // logsDir returns the directory to use for log configuration and
 // buffer storage.
 func logsDir(logf logger.Logf) string {
+	if d := os.Getenv("TS_LOGS_DIR"); d != "" {
+		fi, err := os.Stat(d)
+		if err == nil && fi.IsDir() {
+			return d
+		}
+	}
+
 	// STATE_DIRECTORY is set by systemd 240+ but we support older
 	// systems-d. For example, Ubuntu 18.04 (Bionic Beaver) is 237.
 	systemdStateDir := os.Getenv("STATE_DIRECTORY")

--- a/tstest/integration/integration_test.go
+++ b/tstest/integration/integration_test.go
@@ -639,6 +639,7 @@ func (n *testNode) StartDaemonAsIPNGOOS(t testing.TB, ipnGOOS string) *Daemon {
 		"HTTP_PROXY="+n.env.TrafficTrapServer.URL,
 		"HTTPS_PROXY="+n.env.TrafficTrapServer.URL,
 		"TS_DEBUG_TAILSCALED_IPN_GOOS="+ipnGOOS,
+		"TS_LOGS_DIR="+t.TempDir(),
 	)
 	cmd.Stderr = &nodeOutputParser{n: n}
 	if *verboseTailscaled {
@@ -753,6 +754,7 @@ func (n *testNode) Tailscale(arg ...string) *exec.Cmd {
 	cmd.Dir = n.dir
 	cmd.Env = append(os.Environ(),
 		"TS_DEBUG_UP_FLAG_GOOS="+n.upFlagGOOS,
+		"TS_LOGS_DIR="+n.env.t.TempDir(),
 	)
 	if *verboseTailscale {
 		cmd.Stdout = os.Stdout

